pkg/kubelet/prober/prober_manager_test.go

```diff
@@ -275,24 +282,27 @@ func TestUpdatePodStatus(t *testing.T) {
 
 	// Setup probe "workers" and cached results.
 	m.workers = map[probeKey]*worker{
-		{testPodUID, unprobed.Name, liveness}:       {},
-		{testPodUID, probedReady.Name, readiness}:   {},
-		{testPodUID, probedPending.Name, readiness}: {},
-		{testPodUID, probedUnready.Name, readiness}: {},
-		{testPodUID, terminated.Name, readiness}:    {},
+		{testPodUID, unprobed.Name, liveness}:          {},
+		{testPodUID, probedReady.Name, readiness}:      {},
+		{testPodUID, probedPending.Name, readiness}:    {},
+		{testPodUID, probedUnready.Name, readiness}:    {},
+		{testPodUID, startedNoReadiness.Name, startup}: {},
+		{testPodUID, terminated.Name, readiness}:       {},
 	}
 	m.readinessManager.Set(kubecontainer.ParseContainerID(probedReady.ContainerID), results.Success, &v1.Pod{})
 	m.readinessManager.Set(kubecontainer.ParseContainerID(probedUnready.ContainerID), results.Failure, &v1.Pod{})
+	m.startupManager.Set(kubecontainer.ParseContainerID(startedNoReadiness.ContainerID), results.Success, &v1.Pod{})
 	m.readinessManager.Set(kubecontainer.ParseContainerID(terminated.ContainerID), results.Success, &v1.Pod{})
 
 	m.UpdatePodStatus(testPodUID, &podStatus)
 
 	expectedReadiness := map[probeKey]bool{
-		{testPodUID, unprobed.Name, readiness}:      true,
-		{testPodUID, probedReady.Name, readiness}:   true,
-		{testPodUID, probedPending.Name, readiness}: false,
-		{testPodUID, probedUnready.Name, readiness}: false,
-		{testPodUID, terminated.Name, readiness}:    false,
+		{testPodUID, unprobed.Name, readiness}:           true,
+		{testPodUID, probedReady.Name, readiness}:        true,
+		{testPodUID, probedPending.Name, readiness}:      false,
+		{testPodUID, probedUnready.Name, readiness}:      false,
+		{testPodUID, startedNoReadiness.Name, readiness}: false,
```

Comment:
I'm sorry I might be slow. The test only sets `startup` readiness probe and it returns `Success`. Meaning it's already started. Thus is must be ready. Without this line:

```
m.startupManager.Set(kubecontainer.ParseContainerID(startedNoReadiness.ContainerID), results.Success, &v1.Pod{})
```

it will test that before the result of startup hook is `Success`, `ready` is false.

Reply:
100% right, actually I should test both cases:
NOT started -> false
started -> true
Thanks for pointing this out!

Reply:
(I just need to come back home - will implement during the rebase, if tide can merge one day)